### ISSUE TITLE
BigDecimal::limit has not effect

### ISSUE THREAD

**Issue by nmk** [jruby/jruby#1615]
Using [BigDecimal::limit](http://www.ruby-doc.org/stdlib-1.9.3/libdoc/bigdecimal/rdoc/BigDecimal.html#method-c-limit) has not effect on following computations with BigDecimal instances. This works in MRI 1.9.3 and later.

Compare the following two sessions:

MRI (1.9.3-p545):

```
~ $ irb -r"bigdecimal" -r "bigdecimal/util"
1.9.3-p545 :001 > BigDecimal.limit(3)
 => 0
1.9.3-p545 :002 > 10.to_d / 3.to_d
 => #<BigDecimal:7fd034801d38,'0.333E1',18(36)>
1.9.3-p545 :003 > (10.to_d / 3.to_d).to_f
 => 3.33
1.9.3-p545 :004 > (10.to_d / 3.to_d).to_s("F")
 => "3.33"  
```

Latest JRuby (1.7.11):

```
~ $ irb -r"bigdecimal" -r"bigdecimal/util"
jruby-1.7.11 :001 > BigDecimal.limit(3)
 => 0
jruby-1.7.11 :002 > 10.to_d / 3.to_d
 => #<BigDecimal:136c2147,'0.33333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333E1',200(204)>
jruby-1.7.11 :003 > (10.to_d / 3.to_d).to_f
 => 3.3333333333333335
jruby-1.7.11 :004 > (10.to_d / 3.to_d).to_s("F")
 => "3.3333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333"
```

**Comment by dymaxionuk** [comment]
This bug has been open for 2 years. Surprised nobody else has felt the pain! 

Since #limit is not working for division, I have to post process all my output data, and traverse every field in my data structures and manually round.

This is cumbersome.   Can someone help fix this?  Thanks

**Comment by kares** [comment]
@dymaxionuk you could have tried fixing it than or posting a bounty. it shouldn't be very hard and there's even some code in the comments to start with ...

**Comment by MrBerg** [comment]
I made a pull request for this: https://github.com/jruby/jruby/pull/4830

**Comment by headius** [comment]
Fixed in 9.1.14 and 9.2. See #4830 for details.
